src/fides/api/service/messaging/messaging_crud_service.py

```diff
@@ -81,31 +87,261 @@ def get_messaging_config_by_key(db: Session, key: FidesKey) -> MessagingConfigRe
+        ]
+    return MessagingTemplate.create(db=db, data=data)
+
+
+def delete_template_by_id(db: Session, template_id: str) -> None:
+    logger.info("Finding messaging config with id '{}'", template_id)
+    messaging_template: Optional[MessagingTemplate] = MessagingTemplate.get(
+        db, object_id=template_id
+    )
+    if not messaging_template:
+        raise MessagingConfigNotFoundException(
+            f"No messaging template found with id {template_id}"
+        )
+    templates_with_type = (
+        MessagingTemplate.query(db=db)
+        .filter(MessagingTemplate.type == messaging_template.type)
+        .all()
+    )
+    if len(templates_with_type) <= 1:
+        raise MessagingConfigValidationException(
+            f"Messaging template with id {template_id} cannot be deleted because it is the only template with type {messaging_template.type}"
+        )
+    logger.info("Deleting messaging config with id '{}'", template_id)
+    messaging_template.delete(db)
+
+
+def get_template_by_id(db: Session, template_id: str) -> MessagingTemplate:
+    messaging_template: Optional[MessagingTemplate] = MessagingTemplate.get(
+        db, object_id=template_id
+    )
+    if not messaging_template:
+        raise MessagingConfigNotFoundException(
+            f"No messaging template found with id {template_id}"
+        )
+    return messaging_template
+
+
+def get_default_template_by_type(
+    template_type: str,
+) -> MessagingTemplateWithPropertiesDetail:
+    default_template = DEFAULT_MESSAGING_TEMPLATES.get(template_type)
+    if not default_template:
+        raise MessagingConfigValidationException(
+            f"Messaging template type {template_type} is not supported."
+        )
+    template = MessagingTemplateWithPropertiesDetail(
+        id=None,
+        type=template_type,
+        content=default_template["content"],
+        is_enabled=False,
+        properties=[],
+    )
+    return template
+
+
+# TODO: (PROD-2058) if id is None, we know on FE that this does not exist yet in DB
+def get_all_messaging_templates_summary(
+    db: Session,
+) -> Optional[List[MessagingTemplateWithPropertiesSummary]]:
+    # Retrieve all templates from the database
+    db_templates: Dict[str, List[Dict[str, Any]]] = {}
+    for template in MessagingTemplate.all(db):
+        if db_templates.get(template.type):
+            db_templates[template.type].append(
+                {
+                    "id": template.id,
+                    "type": template.type,
+                    "is_enabled": template.is_enabled,
+                    "properties": template.properties,
+                }
+            )
+        else:
+            db_templates[template.type] = [
+                {
+                    "id": template.id,
+                    "type": template.type,
+                    "is_enabled": template.is_enabled,
+                    "properties": template.properties,
+                }
+            ]
```

Comment:
It looks to me like you're adding the same thing to the array, that the main difference in one conditional, you need to create a key if it doesn't exist.   This is a good use case for a `defaultdict`, something like this:

```python
from collections import defaultdict

db_templates: Dict[str, List[Dict[str, Any]]] = defaultdict(list)
for template in MessagingTemplate.all(db):
    db_templates[template.type].append(
        {
            "id": template.id,
            "type": template.type,
            "is_enabled": template.is_enabled,
            "properties": template.properties,
        }
    )
```